lib/kube/proxy/portforward.go

```diff
@@ -263,10 +263,10 @@ func (h *portForwardProxy) getStreamPair(requestID string) (*httpStreamPair, boo
 func (h *portForwardProxy) monitorStreamPair(p *httpStreamPair, timeout <-chan time.Time) {
 	select {
 	case <-timeout:
-		err := fmt.Errorf("(conn=%v, request=%s) timed out waiting for streams", h.sourceConn, p.requestID)
+		err := fmt.Errorf("(conn=%p, request=%s) timed out waiting for streams", h.sourceConn, p.requestID)
```

Comment:
Do we need to log the pointer address for the connection here (and everywhere else in this method) at all? Just wondering what kind of debugging it can be potentially useful for, other than maybe correlating the addresses to find logs for the same connection. Is there maybe a more useful connection property we can log here which is also more human-readable, say, remote addr?

Reply:
If I log through `portForwardProxy` (w/ `h.Infof()`), I get `log.WithFields` that includes the remote address.  https://github.com/gravitational/teleport/blob/master/lib/kube/proxy/portforward.go#L80-L83

Given that `portForwardProxy.sourceConn` is the connection handler for the remote request, it seems that `log.WithFields` appending remote address should provide enough context to correlate log lines between source connections. 

Proposal:
```golang
// Before
log.Debugf("(conn=%p, request=%s) successfully received error and data streams", h.sourceConn, p.requestID)
// After
h.Debugf("Request: %s, successfully received error and data streams.", p.requestID)
```
Output:
```
DEBU [PROXY:KUB] Request: 5, successfully received error and data streams. addr.remote:24.27.x.x:63537 proxy/portforward.go:270
```

Any idea why we print this error, instead of logging it? 
https://github.com/gravitational/teleport/blob/master/lib/kube/proxy/portforward.go#L266-L267

Reply:
that code is pretty old, but I assume the `printError` call is to give the client some feedback on the problem (instead of only showing it in the server's log)